travis.sh

```diff
@@ -147,7 +157,232 @@ function runFindbugs() {
+              # 1. Line indicates the line at which checkstyle/scalastyle issue exists
+              # 2. Column indicates the column at which checkstyle/scalastyle issue exists
+              # 3. Message is explanation about the issue.
+              # 4. Source is Checkstyle/Scalastyle check which led to the issue.
+              numAttributes = split(issues[issueIdx], attributes, ",");
+              for (attributeIdx = 1; attributeIdx <= numAttributes; attributeIdx++) {
+                lineIdx = index(attributes[attributeIdx], "line=");
+                if (lineIdx > 0) {
+                  line = line "" substr(attributes[attributeIdx], lineIdx + 5);
+                  processingMessage = 0;
+                  continue;
+                }
+                columnIdx = index(attributes[attributeIdx], "column=");
+                if (columnIdx > 0) {
+                  column = column "" substr(attributes[attributeIdx], columnIdx + 7);
+                  processingMessage = 0;
+                  continue;
+                }
+                sourceIdx = index(attributes[attributeIdx], "source=");
+                if (sourceIdx > 0) {
+                  source = source "" substr(attributes[attributeIdx], sourceIdx + 7);
+                  processingMessage = 0;
+                  continue;
+                }
+
+                # Extract message from message attribute. As message can contain commas as well, continue to append to message
+                # till another attribute is encountered.
+                messageIdx = index(attributes[attributeIdx], "message=");
+                if (messageIdx > 0) {
+                  message = message "" substr(attributes[attributeIdx], messageIdx + 8);
+                  processingMessage = 1;
+                } else if (processingMessage == 1) {
+                  message = message "," attributes[attributeIdx];
+                }
+              }
+              # Remove dot from the end of the message.
+              split(message, chars, "");
+              len = length(message);
+              if (chars[len] == ".") {
+                message="" substr(message, 1, len - 1);
+              }
+
+              # Extract last section of source string, separated by dot.
+              numSourceParts = split(source, sourceParts, ".");
+              # Print style information in the desired format
+              printf("\t + %s (%s) at line: %s%s\n", sourceParts[numSourceParts], message, line, ((column == "") ? "." : (" and column: " column ".")));
+            }
+          }'
+        fi
+      fi
+    done
+  )
+  echo "${styleIssuesForChangedFiles}"
+}
+
+############################################################
+# Capitalizes first character of passed string
+#
+# Arguments:
+#   arg1: String whose first character has to be capitalized
+# Returns:
+#   String with first character captialized
+############################################################
+function capitalizeFirstCharOfStr() {
```

Comment:
You can use echo "${1^}" for capitalizing the first character of the String.

Reply:
Are you referring to "${var^}" if the value associated with variable var has to be capitalized?
This works only in Bash 4+. I haven't used Bash 4 constructs anywhere as shell may not be updated to version 4+.

Reply:
Ok. Got it. Will commit the change.